docs/semgrep-supply-chain/transitive-reachability.md

```diff
@@ -0,0 +1,116 @@
+---
+slug: transitive-reachability
+title: Transitive reachability
+hide_title: true
+description: Learn how transitive reachability identifies vulnerabilities introduced by third-party packages.
+tags:
+ - Semgrep Supply Chain
+---
+
+# Transitive reachability (beta)
+
+Semgrep Supply Chains' transitive reachability analysis helps you identify vulnerabilities introduced by third-party packages and their dependencies. 
+
+
+:::info
+This feature is in private beta. To join, contact [support](/support).
+:::
+
+## Supported languages
+
+Semgrep currently performs transitive reachability analysis for JavaScript projects.
+
+## Definitions
+
+For this article:
+- **First-party code**: first-party code refers to your project, which includes the code that Semgrep scans
+- **Third-party code**: third-party code refers to the dependencies that the first-party code calls or imports and then uses
+
+Semgrep uses two overarching categories for project dependencies:
+
+- **Direct**: direct dependencies are those explicitly added by the developer, then called in the first-party code. You can visualize and review direct dependencies using the project's configuration file, such as the lockfile or the manifest file. Semgrep can also generate visual dependency graphs for projects that it scans.
+- **Transitive**: transitive dependencies are those that you include indirectly. For example, this happens when you directly add a dependency, but the dependency that you add then calls another dependency. In other words, a transitive dependency is one that third-party code adds and calls.
+
+## How transitive reachability analysis works
+
+Semgrep's reachability analysis determines whether there's a vulnerability in your codebase by checking your code, specifically the functions used and the function calls made. Then, Semgrep checks if the vulnerability is reachable by checking to see if:
```

Comment:
AFAIK it's not always a function, it can be something else too. @bkettle can probably help clarify more here if we want to add that level of precision.

Reply:
I'll link to the SSC overview